ipset.go

```diff
@@ -332,6 +333,79 @@ type IPSet struct {
 	rr []IPRange
 }
 
+// ParseIPSet parses a comma-separated list of IP ranges into an IPSet.
+// If an invalid IPRange is provided, an empty IPSet and an error are returned.
+// The provided list need not be representative of a true set; ranges may
+// overlap and duplicates are allowed.
+// An empty string produces an empty set.
+func ParseIPSet(s string) (*IPSet, error) {
+	var b IPSetBuilder
+	if len(s) == 0 {
+		return b.IPSet()
+	}
+	rss := strings.Split(s, ",")
+	for _, rs := range rss {
+		r, err := ParseIPRange(rs)
+		if err != nil {
+			return nil, fmt.Errorf("invalid IP range %q: %w", rs, err)
+		}
+		b.AddRange(r)
+	}
+	return b.IPSet()
+}
+
+// MustParseIPSet calls ParseIPSet(s) and panics if an error is encountered.
+func MustParseIPSet(s string) *IPSet {
+	ret, err := ParseIPSet(s)
+	if err != nil {
+		panic(err)
+	}
+	return ret
+}
+
+// String returns a string representation of the IPSet.
+func (s IPSet) String() string {
+	var (
```

Comment:
Please do

```go
var b strings.Builder
var buf []byte
```

Factored variable declarations are fairly rare in practice, so this reads as if you're telling us that b and buf are linked in some important and unusual way. And in this case, spelling out "var" twice takes us from 4 LOC to 2.

Reply:
Added in an err check against the `IPSetBuilder.IPSet` call in the `ParseIPSet` function in order to be explicit about the return value. Also condensed the var declarations in the `String` method. I'd not heard of parens-wrapped var declarations implying an relationship before, but now that you mention it, it makes total sense. Plus it feels easier to read this way somehow.